src/coprocessor/dag/executor/scanner.rs

```diff
@@ -116,34 +134,41 @@ impl<S: Snapshot> Scanner<S> {
             );
         }
 
-        self.seek_key = match (self.scan_mode, self.scan_on) {
-            (ScanMode::Forward, _) => util::prefix_next(&key), // Where is this thing useful?????
-            (ScanMode::Backward, ScanOn::Table) => box_try!(truncate_as_row_key(&key)).to_vec(),
-            (ScanMode::Backward, ScanOn::Index) => key.clone(),
-            _ => unreachable!(),
-        };
+        {
+            let seek_key_slice = match (self.scan_mode, self.scan_on) {
+                (ScanMode::Forward, _) | (ScanMode::Backward, ScanOn::Index) => key.as_slice(),
+                (ScanMode::Backward, ScanOn::Table) => box_try!(truncate_as_row_key(&key)),
```

Comment:
I'm curious why previously in TableScan it needs `truncate_as_row_key`? @AndreMouche